components/802.15.4_RF/stm-s2lp-rf-driver/source/rf_configuration.c

```diff
@@ -181,5 +181,19 @@ uint32_t rf_conf_calculate_deviation(phy_modulation_index_e modulation_index, ui
     return deviation;
 }
 
+uint8_t rf_conf_calculate_pa_level_dbm(int8_t power_dbm)
+{
+    MBED_ASSERT(IS_PAPOWER_DBM(power_dbm));
```

Comment:
No asserts on invalid values.

Please CAP the value between defines as we might not know from stack what is the actual limitations in driver and as those are runtime configurations we might make configuration that starts to ASSERT in field which is not acceptable as the device can't be remotely recovered